internals/models.py

```diff
@@ -319,20 +320,34 @@ def del_none(d):
       del_none(value)
   return d
 
-class DictModel(db.Model):
+class DictModel(ndb.Model):
   # def to_dict(self):
   #   return dict([(p, unicode(getattr(self, p))) for p in self.properties()])
+  saved = False  # class variable provides default value
+
+  @classmethod
+  def _post_get_hook(cls, key, future):
+    obj = future.get_result()
+    if obj is not None:
+      # test needed because post_get_hook is called even if get() fails!
+      obj.saved = True
+
+  def _post_put_hook(self, future):
+    self.saved = True
+
+  def is_saved():
+    return self.saved
```

Comment:
`is_saved()` has been deprecated. These methods may be used to replace as mentioned in [this](https://cloud.google.com/appengine/docs/standard/python/ndb/db_to_ndb) article. 

> model_instance.is_saved() | No direct equivalent; see&nbsp;[Stack Overflow](https://stackoverflow.com/questions/12083254/is-it-possible-to-determine-with-ndb-if-model-is-persistent-in-the-datastore-or/12096066#12096066)&nbsp;for a possible solution.

Reply:
OK that looks fine.